deploy/bin/update-worker-types.js

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env node
+
+const bluebird = require('bluebird');
+const workerTypes = require('../etc/worker-types.json');
+const amis = require('../../docker-worker-amis.json');
+const jsonfile = bluebird.Promise.promisifyAll(require('jsonfile'));
+const utils = require('./worker_type_utils');
+
+/*
+ * Write worker-types configuration to a json file
+ */
+async function backupWorkerTypes(client) {
+  console.log('Creating worker-types backup file.');
+  const names = await client.listWorkerTypes();
+  const workerTypes = names.map(name => {
+    const w = new utils.WorkerType(client, name);
+    return w.workerType();
+  });
+
+  await Promise.all(workerTypes).then(
+    result => jsonfile.writeFileAsync(
+      'worker-types-backup.json',
+      result,
+      { spaces: 2 }
+    )
+  );
+}
+
+function updateWorkerTypes(client) {
+  console.log('Updating worker types...');
+  const p = Object.keys(workerTypes).map(i => {
+    return workerTypes[i].map(async name => {
+      const workerType = new utils.WorkerType(client, name);
+      const wt = await workerType.workerType();
+
+      for (region of wt.regions) {
+        region.launchSpec.ImageId = amis[i][region.region];
+      }
+      console.log(`Updating ${name}`);
+      return workerType.update(wt).then(
+        () => utils.killInstances(workerType)
```

Comment:
I do not feel comfortable with this.  This would be mean that when we deploy a new ami, it'll kill instances currently running a task.  That means many thousands of instances (and jobs failed with claim expired).  Perhaps we can adapt something like what generic-worker is doing and have a deployment id generated.  In between claims of tasks, if the deployment ID has changed from when the worker started, it will shut down.

Reply:
My understanding was that we should kill all instances when deploying new AMIs. If this is not the case, I will remove it. I need to check the generic-worker strategy.